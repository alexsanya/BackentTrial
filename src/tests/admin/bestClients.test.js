const request = require("supertest");
const app = require("../../app.js");
const { seed } = require('../../../scripts/seedDb');
const { HTTP_STATUS_CODES, stripTimestamps } = require('../../common');

describe('Best clients', () => {
  beforeEach(async () => {
    await seed(); //reset database before each test
    console.log('Database been reset');
  });

  it('should provide best clients', async () => {
    const response = await request(app)
      .get(`/admin/best-clients?start=2019/08/09&end=2022/08/11&limit=10`)
      .expect('Content-Type', /json/)
      .expect(HTTP_STATUS_CODES.OK)
    expect(response.body.bestClients.map(item => stripTimestamps(item))).toMatchSnapshot();
  })

  it('should limit output', async () => {
    const response = await request(app)
      .get(`/admin/best-clients?start=2019/08/09&end=2022/08/11&limit=3`)
      .expect('Content-Type', /json/)
      .expect(HTTP_STATUS_CODES.OK)
    expect(response.body.bestClients.map(item => stripTimestamps(item))).toMatchSnapshot();
  })

  it(`should return ${HTTP_STATUS_CODES.BAD_REQUEST} if date format is wrong`, async () => {
    const response = await request(app)
      .get(`/admin/best-clients?start=2019/08/x9&end=2022/08/xx&limit=3`)
      .expect(HTTP_STATUS_CODES.BAD_REQUEST)
  })


})
