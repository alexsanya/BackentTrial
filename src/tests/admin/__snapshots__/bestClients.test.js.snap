// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Best clients should limit output 1`] = `
[
  {
    "fullName": "Ash Kethcum",
    "id": 4,
    "paid": 2020,
  },
  {
    "fullName": "Mr Robot",
    "id": 2,
    "paid": 442,
  },
  {
    "fullName": "Harry Potter",
    "id": 1,
    "paid": 442,
  },
]
`;

exports[`Best clients should provide best clients 1`] = `
[
  {
    "fullName": "Ash Kethcum",
    "id": 4,
    "paid": 2020,
  },
  {
    "fullName": "Mr Robot",
    "id": 2,
    "paid": 442,
  },
  {
    "fullName": "Harry Potter",
    "id": 1,
    "paid": 442,
  },
  {
    "fullName": "John Snow",
    "id": 3,
    "paid": 200,
  },
]
`;
